pipelines/matrix/src/matrix/pipelines/run_comparison/evaluations.py

```diff
@@ -0,0 +1,87 @@
+import abc
+from typing import List
+
+import pandas as pd
+import polars as pl
+import pyspark.sql as ps
+
+
+class ComparisonEvaluation(abc.ABC):
+    """Abstract base class for run-comparison evaluations."""
+
+    @abc.abstractmethod
+    def evaluate(self, matrices: List[ps.DataFrame]) -> pd.DataFrame:
+        pass
+
+
+class RecallAtN(ComparisonEvaluation):
+    """Recall@N evaluation"""
```

Comment:
Would probably allow for setting some of the args of `give_recall_at_n` though the object initialiser.